### PR TITLE
Fix dynamic configuration watch implementation current value not null when the config is deleted.

### PR DESCRIPTION
**Pull request by wankai123**
### Fix <bug description or the bug issue number or bug issue link>
- [ ] Add a unit test to verify that the fix works.
- [X] Explain briefly why the bug exists and how to fix it.
   
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

In `ConfigWatcherRegister.singleConfigsSync()`, when config is deleted and `watcher.value()` is not null, core would notify watcher every syncPeriod time. 
``` java
ConfigChangeWatcher watcher = holder.getWatcher();
                String newItemValue = item.getValue();
                if (newItemValue == null) {
                    if (watcher.value() != null) {
                        // Notify watcher, the new value is null with delete event type.
                        watcher.notify(
                            new ConfigChangeWatcher.ConfigChangeEvent(null, ConfigChangeWatcher.EventType.DELETE));
                    } else {
                        // Don't need to notify, stay in null.
                    }
```

### CODE REVIEW

**Review by wu-sheng — APPROVED** [review]

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/ApdexThresholdConfig.java [hunk 1]
```diff
@@ -90,7 +89,9 @@ private synchronized void activeSetting(String config) {
             log.debug("Updating using new static config: {}", config);
         }
         rawConfig = config;
-        updateConfig(new StringReader(config));
+        if (config != null) {
+            updateConfig(new StringReader(config));
+        }
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/ApdexThresholdConfig.java:94]: `config == ""` was an signal to "reset" the config, with this new `if` statement, the config might never been reset even the dynamic config is deleted?

Comment (wankai123) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/ApdexThresholdConfig.java]: fixed

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/logging/LoggingConfigWatcher.java [hunk 2]
```diff
@@ -63,14 +64,6 @@ public void notify(final ConfigChangeEvent value) {
...
             return;
         }
-        StringBuilder builder = new StringBuilder();
-        ctx.getConfiguration().getLoggers().forEach((loggerName, config) -> {
-            builder.append(Strings.isNullOrEmpty(loggerName) ? "Root" : loggerName)
-                   .append(":")
-                   .append(config.getLevel())
-                   .append(",");
-        });
-        this.content = builder.toString();
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/logging/LoggingConfigWatcher.java:73]: @hanahmily This was a bug of old codes, which cause endless loop of update notification.

File: oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/UninstrumentedGatewaysConfig.java [hunk 3]
```diff
@@ -62,13 +61,15 @@ private void activeSetting(String config) {
             log.debug("Updating using new static config: {}", config);
         }
         this.settingsString.set(config);
-        onGatewaysUpdated(parseGatewaysFromYml(config));
+        if (config != null) {
+            onGatewaysUpdated(parseGatewaysFromYml(""));
+        }
```
Comment (kezhenxu94) [inline oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/UninstrumentedGatewaysConfig.java:66]: You might mean `if (config == null)` update with `""`? The same in other places in this PR

```suggestion
        if (config == null) {
            onGatewaysUpdated(parseGatewaysFromYml(""));
        }
```

Comment (wankai123) [reply oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/UninstrumentedGatewaysConfig.java]: yes, mistake that cause ci failed

**Review by kezhenxu94 — DISMISSED** [review]
LGTM, just one nit

File: oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/UninstrumentedGatewaysConfig.java [hunk 4]
```diff
@@ -62,13 +61,17 @@ private void activeSetting(String config) {
             log.debug("Updating using new static config: {}", config);
         }
         this.settingsString.set(config);
-        onGatewaysUpdated(parseGatewaysFromYml(config));
+        if (config != null) {
+            onGatewaysUpdated(parseGatewaysFromYml(config));
+        } else {
+            onGatewaysUpdated(parseGatewaysFromYml(""));
+        }
```
[the remark this answers is not in the document]
Comment (kezhenxu94) [reply oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/UninstrumentedGatewaysConfig.java]: The same as other places

**Review by kezhenxu94 — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
@JaredTan95 when you do merging in the next time, please make the commit logs clean.

```
Fix dynamic configuration watch implementation current value not null…
… when the config is deleted. (#7606)

* Fix dynamic configuration watch implementation current value not null when the config is deleted.

* add changes.md

* fix LoggingConfigWatcher watch.vale, and when value = null ,config not reset.

* fix

* fix style

* Update oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/UninstrumentedGatewaysConfig.java

Co-authored-by: kezhenxu94 <kezhenxu94@apache.org>

* use Strings.nullToEmpty

Co-authored-by: 吴晟 Wu Sheng <wu.sheng@foxmail.com>
Co-authored-by: kezhenxu94 <kezhenxu94@apache.org>
```
This is how it looks like now. Too many unnecessary things.

**Pull request merged**